internal/db/db_sqlite.go

```diff
@@ -356,3 +356,59 @@ func (s *Sqlite) FindUser(ctx context.Context, id string) (*snips.User, error) {
 
 	return user, nil
 }
+
+func (s *Sqlite) LatestPublicFiles(ctx context.Context, page int, limit int) ([]*snips.File, error) {
+	const query = `
+		SELECT
+			id,
+			created_at,
+			updated_at,
+			size,
+			private,
+			type,
+			user_id
+		FROM files
+		WHERE private = 0
+		ORDER BY created_at DESC
+		LIMIT ?, ?
+	`
+
+	// just a precaution
+	if limit < 0 {
+		limit = 0
+	}
+
+	page -= 1
+	if page < 0 {
+		page = 0
+	}
+
+	files := make([]*snips.File, 0)
+	rows, err := s.QueryContext(ctx, query, page*limit, limit)
+	if err != nil {
+		return files, err
+	}
+
```

Comment:
```suggestion
	rows, err := s.QueryContext(ctx, query, page*limit, limit)
	if err != nil {
		return files, err
	}
	defer rows.Close()

```

I prefer to have the [extra defer in there](https://github.com/golang/go/issues/46863#issue-926578255)